frontend/projects/ui/src/app/components/toast-container/offline-toast/offline-toast.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core'
+import { Observable, of } from 'rxjs'
+import {
+  debounceTime,
+  distinctUntilChanged,
+  map,
+  switchMap,
+} from 'rxjs/operators'
+
+import { AuthService } from 'src/app/services/auth.service'
+import {
+  ConnectionFailure,
+  ConnectionService,
+} from 'src/app/services/connection.service'
+
+export interface OfflineMessage {
+  readonly message: string
+  readonly link?: string
+}
+
+// TODO: Update according to new offline statuses
+// Watch for connection status
+@Injectable({ providedIn: 'root' })
+export class OfflineToastService extends Observable<OfflineMessage | null> {
+  private readonly stream$ = this.authService.isVerified$.pipe(
+    switchMap(verified =>
+      verified
+        ? this.connectionService
+            .watchFailure$()
+            .pipe(distinctUntilChanged(), debounceTime(500), map(getMessage))
+        : of(null),
+    ),
+  )
+
+  constructor(
+    private readonly authService: AuthService,
+    private readonly connectionService: ConnectionService,
+  ) {
+    super(subscriber => this.stream$.subscribe(subscriber))
+  }
+}
+
+function getMessage(failure: ConnectionFailure): OfflineMessage | null {
```

Comment:
This is probably the only place where this code differs from your source refactor, @MattDHill. Hence `TODO` on this file.